File: src/main/java/org/simbrain/network/gui/actions/dl4j/AddMultiLayerNet.java
package org.simbrain.network.gui.actions.dl4j;

import org.simbrain.network.gui.NetworkPanel;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class AddMultiLayerNet extends AbstractAction {


    /**
     * Network panel.
     */
    private final NetworkPanel networkPanel;

    /**
     * Create a new DL4J Multilayer net to the specified network panel.
     *
     * @param networkPanel network panel, must not be null
     */
    public AddMultiLayerNet(final NetworkPanel networkPanel) {
        super("Add Multilayer Network...");

        if (networkPanel == null) {
            throw new IllegalArgumentException("networkPanel must not be null");
        }

        this.networkPanel = networkPanel;
        putValue(SHORT_DESCRIPTION, "Add a Dl4J Multi Layer Network to the network");

    }

    @Override
    public void actionPerformed(final ActionEvent event) {
        networkPanel.showMultiLayerNetworkCreationDialog();
    }

}

File: src/main/kotlin/org/simbrain/network/gui/actions/dl4j/AddMultiLayerNet.kt
package org.simbrain.network.gui.actions.dl4j

import org.simbrain.network.gui.NetworkPanel
import org.simbrain.network.gui.showMultiLayerNetworkCreationDialog
import java.awt.event.ActionEvent
import javax.swing.AbstractAction
import javax.swing.Action

class AddMultiLayerNet(val networkPanel: NetworkPanel) : AbstractAction("Add Multilayer Network...") {

    override fun actionPerformed(event: ActionEvent) {
        networkPanel.showMultiLayerNetworkCreationDialog()
    }

    init {
        putValue(Action.SHORT_DESCRIPTION, "Add a Dl4J Multi Layer Network to the network")
    }
}